janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -264,6 +265,29 @@
      */
     public static final double DEFAULT_GEO_DIST_ERROR_PCT = 0.025;
 
+    private static final String DELETION_SCRIPT = String.join(" ", "",
+            "for (field in params.fields) {",
+            "    if (field.cardinality == 'SINGLE') {",
+            "        ctx._source.remove(field.name);",
+            "    } else if (ctx._source.containsKey(field.name)) {",
+            "        def fieldIndex = ctx._source[field.name].indexOf(field.value);",
```

Comment:
I just compared this string to the one created in `getDeletionScript` and noticed that `field.value` is converted there with `convertToJsType` for this `indexOf` operation. Any reason why this isn't necessary any more?

Reply:
@FlorianHockmann `convertToJsType` is used to serialize value to `String` which is going to be done by ES client itself since we don't generate dynamic scripts anymore. But I've re-used `convertToEsType` only (which is called inside `convertToJsType`) for backward compatibility.